File: app/src/main/java/com/blog/demo/image/canvas/CanvasBitmapActivity.java
package com.blog.demo.image.canvas;

import android.app.Activity;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Path;
import android.os.Bundle;
import android.widget.ImageView;

import androidx.annotation.Nullable;

import com.blog.demo.R;

public class CanvasBitmapActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_image_canvas_bitmap);

        ImageView imageView = findViewById(R.id.image_view);

        Bitmap bitmap = Bitmap.createBitmap(600, 800, Bitmap.Config.ARGB_8888);
        Canvas canvas = new Canvas(bitmap);
        draw(canvas);

        imageView.setImageBitmap(bitmap);
    }

    private void draw(Canvas canvas) {
        final String text = "This is a text";
        Paint redPaint = new Paint();
        redPaint.setColor(Color.RED);
        redPaint.setStyle(Paint.Style.FILL);
        redPaint.setTextSize(20);
        redPaint.setAntiAlias(true); // 设置是否使用抗锯齿功能

        canvas.drawText(text, 50, 30, redPaint);
        canvas.drawText(text, 2, 14, 50, 55, redPaint);

        Paint paint = new Paint(redPaint);
        paint.setStyle(Paint.Style.STROKE);
        Path path = new Path();
        path.moveTo(50, 100);
        path.lineTo(300, 200);
        canvas.drawPath(path, paint);
        canvas.drawTextOnPath(text, path, 80, 20, redPaint);
    }


}

File: app/src/main/java/com/blog/demo/image/canvas/CanvasBitmapActivity.kt
package com.blog.demo.image.canvas

import android.app.Activity
import android.graphics.*
import android.os.Bundle
import android.widget.ImageView
import com.blog.demo.R

class CanvasBitmapActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_image_canvas_bitmap)

        val imageView: ImageView = findViewById(R.id.image_view)
        val bitmap = Bitmap.createBitmap(600, 800, Bitmap.Config.ARGB_8888)
        val canvas = Canvas(bitmap)
        draw(canvas)
        imageView.setImageBitmap(bitmap)
    }

    private fun draw(canvas: Canvas) {
        val text = "This is a text"
        val redPaint = Paint()
        redPaint.color = Color.RED
        redPaint.style = Paint.Style.FILL
        redPaint.textSize = 20f
        redPaint.isAntiAlias = true // 设置是否使用抗锯齿功能
        canvas.drawText(text, 50f, 30f, redPaint)
        canvas.drawText(text, 2, 14, 50f, 55f, redPaint)

        val paint = Paint(redPaint)
        paint.style = Paint.Style.STROKE

        val path = Path()
        path.moveTo(50f, 100f)
        path.lineTo(300f, 200f)

        canvas.drawPath(path, paint)
        canvas.drawTextOnPath(text, path, 80f, 20f, redPaint)
    }

}